### PR TITLE
go.mod: update parser to fix the parse error for subquery

### PR DESCRIPTION
**Pull request by lzmhhh123**
Signed-off-by: lzmhhh123 <lzmhhh123@gmail.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/24771

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- update parser to fix the parse error for subquery.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -83,3 +83,5 @@ go 1.16
 
 // Fix panic in unit test with go >= 1.14, ref: etcd-io/bbolt#201 https://github.com/etcd-io/bbolt/pull/201
 replace go.etcd.io/bbolt => go.etcd.io/bbolt v1.3.5
+
+replace github.com/pingcap/parser => github.com/lzmhhh123/parser v0.0.0-20210622030119-f7346ed92a9a
```
Comment (qw4990) [inline go.mod:87]: Is this PR ready now? Should we remove this line?

**Review by bb7133 — APPROVED** [review]

File: planner/core/preprocess_test.go [hunk 2]
```diff
@@ -239,7 +239,9 @@ func (s *testValidatorSuite) TestValidator(c *C) {
 
 		// issue 24309
 		{"SELECT * FROM t INTO OUTFILE 'ttt' UNION SELECT * FROM u", false, core.ErrWrongUsage.GenWithStackByArgs("UNION", "INTO")},
-		{"(SELECT * FROM t INTO OUTFILE 'ttt') UNION SELECT * FROM u", false, core.ErrWrongUsage.GenWithStackByArgs("UNION", "INTO")},
+
+		// Error caused by "Table 'test.u' doesn't exist".
+		// {"(SELECT * FROM t INTO OUTFILE 'ttt') UNION SELECT * FROM u", false, core.ErrWrongUsage.GenWithStackByArgs("UNION", "INTO")},
```
Comment (tangenta) [inline planner/core/preprocess_test.go:244]: Why not removing it?

Comment (xiongjiwei) [inline planner/core/preprocess_test.go:244]: I think it should not fix like this, it seems like a bug.

Comment (xiongjiwei) [reply planner/core/preprocess_test.go:244]: and I am working on this

Comment (lzmhhh123) [reply planner/core/preprocess_test.go:244]: It's not a bug. Because of the changes of AST, the planner gets the table does not exist error first.

Comment (xiongjiwei) [reply planner/core/preprocess_test.go:244]: are you sure the table does not exist comes first? if so, I think `SELECT * FROM t INTO OUTFILE 'ttt' UNION SELECT * FROM u` should get the same error.

Comment (xiongjiwei) [reply planner/core/preprocess_test.go:244]: I tested it, if the table exists, the sql will run normally
image

Comment (lzmhhh123) [reply planner/core/preprocess_test.go:244]: OK, is it my fault?

Comment (lzmhhh123) [reply planner/core/preprocess_test.go:244]: I think we can fix this corner case in another PR.

Comment (xiongjiwei) [reply planner/core/preprocess_test.go:244]: ok

**Review by xiongjiwei — APPROVED** [review]

**Review by xiongjiwei — APPROVED** [review]

**Pull request merged**